### PR TITLE
Refactoring mainly in alerter component

### PR DESCRIPTION
**Pull request by matthew-peacock-ig**

### CODE REVIEW

File: alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/InitialTableScanner.java [hunk 1]
```diff
@@ -49,7 +42,7 @@
 
 	@EventListener(ApplicationReadyEvent.class)
 	void scanForTriggers() {
-		log.info("scanning entities scanChunk={} ***********************", scanChunk);
+		log.info("scanning entities for triggers scanChunk={} ***********************", scanChunk);
```
Comment (steliospaps) [inline alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/InitialTableScanner.java]: for now we are scanning triggers, but in reality we scan everything in the table. So I think the 'for triggers' here might be misleading. Actually the name of the function should be changed.

File: alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/InitialTableScanner.java [hunk 2]
```diff
@@ -70,7 +63,7 @@ void scanForTriggers() {
 								);
 
 					} else {
-						log.info("skipping");
+						log.info("ignoring (non trigger row)");
```
Comment (steliospaps) [inline alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/InitialTableScanner.java:66]: cool. when we start processing INSTRUMENT entities, will we update this to that?

File: alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/websocket/client/WebsocketClient.java [hunk 3]
```diff
@@ -122,6 +121,7 @@ public void shutdown() {
 				.log("ws-output",Level.FINEST,SignalType.ON_NEXT)//
 				.log("ws-output",Level.INFO,SignalType.ON_SUBSCRIBE,SignalType.CANCEL,SignalType.ON_COMPLETE,SignalType.ON_ERROR,SignalType.AFTER_TERMINATE)//
 				.map(toSend->ws.textMessage(toSend)))
+				.doOnTerminate(() -> log.warn("will retry connection in {} seconds", retryConnectionInterval.getSeconds()))
```
Comment (steliospaps) [inline alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/websocket/client/WebsocketClient.java:124]: doOnTerminate does not hande cancel consider:
doFinally(Consumer<SignalType> onFinally)
Add behavior (side-effect) triggered after the Flux terminates for any reason, including cancellation.

File: alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/websocket/client/WebsocketClient.java [hunk 4]
```diff
@@ -147,6 +147,9 @@ public void shutdown() {
 			case "Quote":
 				publishPrices(mapper.treeToValue(jsonNode, Quote.class));
 				return Flux.empty();
+			case "NegotiationReject":
+				log.warn("Session failed to logon, reason={}", jsonNode.get("Reason"));
+				return Flux.empty();
```
Comment (steliospaps) [inline alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/websocket/client/WebsocketClient.java:152]: does it make it exit? or we rely on the timer to kick us out?

**Review by steliospaps — CHANGES_REQUESTED** [review]

File: alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/InitialTableScanner.java (hunk 2 above)
Comment (matthew-peacock-ig) [reply alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/InitialTableScanner.java:66]: Yeah we can refactor this part if we want to process instruments on initial startup

File: alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/InitialTableScanner.java (hunk 1 above)
Comment (matthew-peacock-ig) [reply alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/InitialTableScanner.java]: will update method name and log line

File: alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/websocket/client/WebsocketClient.java (hunk 4 above)
Comment (matthew-peacock-ig) [reply alerter-component/src/main/java/io/github/steliospaps/ighackathon/instrumentpricealerter/alertercomponent/websocket/client/WebsocketClient.java:152]: no this doesnt make it exit, it handles this like heartbeat messages. just returns an empty flux (no elements to send back) in response to the NegotiationReject. The server kicks out a client immediately after this reject message.

**Review by steliospaps — APPROVED** [review]

**Pull request merged**